src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -170,25 +179,26 @@ public IDisposable On(string methodName, Type[] parameterTypes, Func<object[], o
                     }
                 }, irq);
             }
-
             return channel;
         }
 
-        public async Task<object> InvokeAsync(string methodName, Type returnType, object[] args, CancellationToken cancellationToken = default(CancellationToken)) =>
+        public async Task<object> InvokeAsync(string methodName, Type returnType, object[] args, CancellationToken cancellationToken = default) =>
              await InvokeAsyncCore(methodName, returnType, args, cancellationToken).ForceAsync();
 
         private async Task<object> InvokeAsyncCore(string methodName, Type returnType, object[] args, CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
             var irq = InvocationRequest.Invoke(cancellationToken, returnType, GetNextId(), _loggerFactory, this, out var task);
             await InvokeCore(methodName, irq, args, nonBlocking: false);
             return await task;
         }
 
-        public async Task SendAsync(string methodName, object[] args, CancellationToken cancellationToken = default(CancellationToken)) =>
+        public async Task SendAsync(string methodName, object[] args, CancellationToken cancellationToken = default) =>
             await SendAsyncCore(methodName, args, cancellationToken).ForceAsync();
 
         private Task SendAsyncCore(string methodName, object[] args, CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
I think this is not good enough. In general this cancellation token is not very useful. The only thing it cancels is writing to the channel. The way to do this is to cancel the actual send from the transport. I think it is quite a big change and should be done separately.